### ISSUE TITLE
SchemaRegistry/Protobuf: topic deserialization doesn't work

### ISSUE THREAD

**Issue by demimurg** [provectus/kafka-ui#3483]
**Describe the bug** (Actual behavior)
Despite the fact that the protobuf schema is created for topic (\<topic\>-value pattern), deserialization in browse window doesn't work

**Expected behavior**
Messages in topic showed in human readable format according to schema registry

**Set up**
Docker Image v0.0.6 (also tried v0.5.2, master, latest)

**Steps to Reproduce**
1. Run kafka, schema-registry and kafkaui
```yaml
version: "3"
volumes:
    kafka_data:
        driver: local
services:
    kafka:
        image: bitnami/kafka:latest
        ports:
            - "9092:9092"
        volumes:
            - "kafka_data:/bitnami/kafka"
        environment:
            # all parameters required for kraft
            - KAFKA_ENABLE_KRAFT=yes
            - KAFKA_CFG_PROCESS_ROLES=broker,controller
            - KAFKA_CFG_CONTROLLER_LISTENER_NAMES=CONTROLLER
            - KAFKA_CFG_LISTENERS=PLAINTEXT://:9092,CONTROLLER://:9093
            - KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP=CONTROLLER:PLAINTEXT,PLAINTEXT:PLAINTEXT
             # you should add "127.0.0.1 kafka" to /etc/hosts to connect outside from docker
            - KAFKA_CFG_ADVERTISED_LISTENERS=PLAINTEXT://kafka:9092
            - KAFKA_BROKER_ID=1
            - KAFKA_CFG_CONTROLLER_QUORUM_VOTERS=1@127.0.0.1:9093
            - ALLOW_PLAINTEXT_LISTENER=yes
    schema-registry:
        image: confluentinc/cp-schema-registry:latest
        ports:
            - "8081:8081"
        depends_on:
            - kafka
        environment:
            SCHEMA_REGISTRY_KAFKASTORE_BOOTSTRAP_SERVERS: PLAINTEXT://kafka:9092
            SCHEMA_REGISTRY_HOST_NAME: schema-registry
    kafkaui:
        image: provectuslabs/kafka-ui:v0.6.0
        ports:
            - "8080:8080"
        depends_on:
            - kafka
            - schema-registry
        environment:
            KAFKA_CLUSTERS_0_BOOTSTRAPSERVERS: kafka:9092
            KAFKA_CLUSTERS_0_SCHEMAREGISTRY: http://schema-registry:8081
```
2. Add protobuf schema for test topic in schema registry
3. Send protobuf messages from your client to kafka topic (i used go app)
4. Look at test topic messages with `Value Serde = SchemaRegistry` selected

**Screenshots**
<img width="1351" alt="image" src="https://user-images.githubusercontent.com/40828320/224482138-41ef0365-09a9-4647-b45c-3890efb6e060.png">
<img width="1305" alt="image" src="https://user-images.githubusercontent.com/40828320/224482326-6d3c2984-53b9-4828-8f8b-6d9da205941b.png">

**Additional context**
Make it work only with [Protobuf Serde](https://github.com/provectus/kafka-ui/blob/master/documentation/guides/Serialization.md) without SchemaRegistry

**Comment by iliax** [comment]
Hello @demimurg , thank you for the issue.

To exclude the possibility that you use different serialization in your app - can you please produce message to the topic via kafka-ui (using `SchemaRegistry` serde) ? To do it click on Produce button, make sure `SchemaRegistry` serde chosenva for value, enter message (ex `{ "simple_field" : "string here", "int_field": 123 }` .

**Comment by demimurg** [comment]
I tried this. Message from produce window looks good in kafkaui but my go app consumer can't deserialize this and fails with error. I am using "google.golang.org/protobuf/proto", so i think there shouldn't be mistake

**Comment by iliax** [comment]
@demimurg
Ok, in this case you probably not using SchemaRegistry properly with your application.
 SchemaRegistry-style serialization has specific message encoding rule (1 byte (zero magic byte) + 4 bytes (schema id in SR) + serialized message bytes). Using this rule consumer can figure out (by schema id) what subject it need to use when deserializing polled message. In your case (considering string representation of your messages)  - you probably just serialized messages into protobuf schema and send it to kafka (without SR-related prefixes). In this case, you don't have SR benefits.
 
Please see details [here](https://docs.confluent.io/platform/current/schema-registry/serdes-develop/index.html#formats-serializers-and-deserializers). 

BTW, kafka-ui follows SR-related ser/deser rules (as it is described in doc), so you can test your app with it

**Comment by Jaypov** [comment]
@demimurg

Did you ever figure out how to serialise prtobuf messages properly? i have been struggling with this issue for a couple of days.  am using the very out of date "@kafkajs/confluent-schema-registry". I did a bunch of research before this post and understand what @iliax is saying. I tried to make my own serialise, but so far have note been successful. Any hits or resources explaining how to implement a steriliser would be great. thanks
